OpenSSL/SSL.py

```
@@ -809,6 +812,92 @@ def wrapper(ssl, alert, arg):
         _lib.SSL_CTX_set_tlsext_servername_callback(
             self._context, self._tlsext_servername_callback)
 
+
+    def set_npn_advertise_callback(self, callback):
+        """
+        Specify a callback function that will be called when offering `Next
+        Protocol Negotiation
+        <https://technotes.googlecode.com/git/nextprotoneg.html>`_ as a server.
+
+        :param callback: The callback function.  It will be invoked with one
+            argument, the Connection instance.  It should return a list of
+            bytestrings representing the advertised protocols, like
+            ``[b'http/1.1', b'spdy/2']``.
+        """
+        @wraps(callback)
+        def wrapper(ssl, out, outlen, arg):
+            conn = Connection._reverse_mapping[ssl]
+            protos = callback(conn)
```

What worries me is debuggability. Swallowing exceptions here will make it really tricky for developers to spot their mistakes. Ideally I'd call `log.exception` before swallowing it, but I'm a bit nervous about adding logging where there previously was none.

Do you have any ideas?